File: Code/Python/LAB10_04.py
class Calculator:
    # ฟังก์ชันสำหรับการบวก
    def add(self, a, b):
        return a + b

class SubtractionCalculator(Calculator):
    # ฟังก์ชันสำหรับการลบ
    def subtract(self, a, b):
        return a - b

class MultiplicationCalculator(SubtractionCalculator):
    # ฟังก์ชันสำหรับการคูณ
    def multiply(self, a, b):
        return a * b

# ฟังก์ชันหลัก
if __name__ == "__main__":
    # สร้างออบเจกต์จากคลาส MultiplicationCalculator
    mul_calc = MultiplicationCalculator()

    # การบวก
    print(f"Addition (10 + 5): {mul_calc.add(10, 5)}")

    # การลบ
    print(f"Subtraction (10 - 5): {mul_calc.subtract(10, 5)}")

    # การคูณ
    print(f"Multiplication (10 * 5): {mul_calc.multiply(10, 5)}")

File: Code/Cplus/LAB10_04/LAB10_04.cpp
#include <iostream>
using namespace std;

// คลาสแม่: Calculator
class Calculator {
public:
    // ฟังก์ชันสำหรับการบวก
    int add(int a, int b) {
        return a + b;
    }
};

// คลาสลูก: SubtractionCalculator
class SubtractionCalculator : public Calculator {
public:
    // ฟังก์ชันสำหรับการลบ
    int subtract(int a, int b) {
        return a - b;
    }
};

// คลาสลูก: MultiplicationCalculator
class MultiplicationCalculator : public SubtractionCalculator {
public:
    // ฟังก์ชันสำหรับการคูณ
    int multiply(int a, int b) {
        return a * b;
    }
};

// ฟังก์ชันหลัก
int main() {
    // สร้างออบเจกต์จากคลาส MultiplicationCalculator
    MultiplicationCalculator mulCalc;

    // การบวก
    cout << "Addition (10 + 5): " << mulCalc.add(10, 5) << endl;

    // การลบ
    cout << "Subtraction (10 - 5): " << mulCalc.subtract(10, 5) << endl;

    // การคูณ
    cout << "Multiplication (10 * 5): " << mulCalc.multiply(10, 5) << endl;

    return 0;
}
